crates/rune/src/diagnostics/warning.rs

```diff
@@ -114,6 +117,14 @@ pub enum WarningDiagnosticKind {
         /// Span where the semi-colon is.
         span: Span,
     },
+    UsedDeprecated {
+        /// The span which is deprecated
+        span: Span,
+        /// The context in which it is used.
+        context: Option<Span>,
+        /// The defined deprecation message
+        message: Box<str>,
```

Comment:
The message doesn't have to be stored in here, it can instead be looked up in the `Context` or `Unit` when diagnostics is being emitted.